rd_ui/app/views/query.html

```diff
@@ -143,6 +143,19 @@
                 <select ng-disabled="!isQueryOwner" ng-model="query.data_source_id" ng-change="updateDataSource()" ng-options="ds.id as ds.name for ds in dataSources"></select>
             </p>
 
+            <p>
+                <i class="fa fa-lock"></i>
+                <span class="text-muted">Access Groups</span>
+                <div>
+                  <ui-select multiple ng-model="query.access_groups">
+                      <ui-select-match placeholder="Select access groups">{{$item}}</ui-select-match>
+                      <ui-select-choices repeat="group in availableGroups | filter:$select.search">
+                        {{group}}
+                      </ui-select-choices>
+                  </ui-select>
```

Comment:
Sure! Though there is an issue with ui-select (I've documented in code), it can be done by applying a little bit hack :smile: